client/signup/steps/domains/index.jsx

```diff
@@ -248,8 +248,20 @@ export class RenderDomainsStep extends Component {
 		const productToRemove = this.props.cart.products.find(
 			( product ) => product.meta === domain_name
 		);
+
+		this.setState( ( prevState ) => ( {
+			domainRemovalQueue: [
+				...prevState.domainRemovalQueue,
+				{ meta: productToRemove.meta, productSlug: productToRemove.product_slug },
+			],
+		} ) );
```

Comment:
Here we're ensuring multiple requests for auto-deletion on failed domains won't fail.